pkg/scheduling/requirements.go

```diff
@@ -220,15 +220,30 @@ func editDistance(s, t string) int {
 }
 
 func labelHint(r Requirements, key string, allowedUndefined sets.Set[string]) string {
```

Comment:
There are test cases for checking the error that's returned from this `labelHint` function. Can we add a few test cases to validate this new behavior?

Reply:
sure